src/buildApplication/buildVercelOutput.ts

```diff
@@ -60,3 +61,24 @@ async function runVercelBuild(): Promise<void> {
 		});
 	});
 }
+
+/**
+ * Vercel and Next.js generate *private* files for telemetry purposes that are accessible as static assets (`.vercel/output/static/_next/__private/...`).
+ *
+ * The routing system for the build output *should* prevent these files from being accessible, but if someone were to exclude all static assets in an `_routes.json` file, they would be accessible.
+ *
+ * We do not need these files, nor do we want to run the risk of them being available. Therefore, we should purge them instead of uploading them to Cloudflare Pages.
+ */
+export async function purgePrivateFiles(): Promise<void> {
+	const nextDir = resolve('.vercel/output/static/_next');
+	const privateDir = join(nextDir, '__private');
+
+	if (await validateDir(privateDir)) {
+		await rm(privateDir, { recursive: true, force: true });
+
+		// Remove the `_next` directory if it's now empty
+		if ((await readdir(nextDir)).length === 0) {
+			await rm(nextDir, { recursive: true, force: true });
+		}
```

Comment:
is this needed? would the empty dir there cause any issue?

Reply:
(also instead of `readdir` + `rm` I think you could use [`rmdir`](https://www.geeksforgeeks.org/node-js-fs-rmdir-method/) with a try catch (since it throws if the directory is not empty)?)

Reply:
> is this needed? would the empty dir there cause any issue?

The empty directory wouldn't cause issues, but it's entirely unnecessary to keep if it's empty.

> (also instead of `readdir` + `rm` I think you could use [`rmdir`](https://www.geeksforgeeks.org/node-js-fs-rmdir-method/) with a try catch (since it throws if the directory is not empty)?)

Good call.